docs/contributing-to-airbyte/migrate-connector-to-poetry.md

```diff
@@ -0,0 +1,108 @@
+# Migrate connector to use Poetry and base Docker Image
+
+1. Remove the `Dockerfile` and `.dockerignore`
+2. Include the `baseImage` to the `metadata.yaml` file. For now, look for `baseImage` to find examples:
+```yaml
+data:
+  allowedHosts:
+    hosts:
+      - *
+  connectorBuildOptions:
+    baseImage: docker.io/airbyte/python-connector-base:1.2.0@sha256:c22a9d97464b69d6ef01898edf3f8612dc11614f05a84984451dde195f337db9
+  connectorSubtype: api
+  connectorType: source
+  dockerImageTag: 0.1.0
+```
+3. Create a file called `pyproject.toml` and transfer all lib used in `setup.py` to this file
+
+* `setup.py`
+```python
+from setuptools import find_packages, setup
+
+MAIN_REQUIREMENTS = [
+    "airbyte-cdk~=0.1",
+]
+
+TEST_REQUIREMENTS = [
+    "requests-mock~=1.9.3",
+    "pytest~=6.2",
+    "pytest-mock~=3.6.1",
+    "connector-acceptance-test",
+]
+
+setup(
+    entry_points={
+        "console_scripts": [
+            "source-openweather=source_openweather.run:run",
+        ],
+    },
+    name="source_openweather",
+    description="Source implementation for Openweather.",
+    author="Airbyte",
+    author_email="contact@airbyte.io",
+    packages=find_packages(),
+    install_requires=MAIN_REQUIREMENTS,
+    package_data={
+        "": [
+            # Include yaml files in the package (if any)
+            "*.yml",
+            "*.yaml",
+            # Include all json files in the package, up to 4 levels deep
+            "*.json",
+            "*/*.json",
+            "*/*/*.json",
+            "*/*/*/*.json",
+            "*/*/*/*/*.json",
+        ]
+    },
+    extras_require={
+        "tests": TEST_REQUIREMENTS,
+    },
+)
+```
+
+* `pyproject.toml` 
+```toml
+[build-system]
+requires = [ "poetry-core>=1.0.0",]
+build-backend = "poetry.core.masonry.api"
+
+[tool.poetry]
+version = "0.2.1"
+name = "source-openweather"
+description = "Source implementation for Open Weather."
+authors = [ "Airbyte <contact@airbyte.io>",]
+license = "MIT"
+readme = "README.md"
+documentation = "https://docs.airbyte.com/integrations/sources/orb"
+homepage = "https://airbyte.com"
+repository = "https://github.com/airbytehq/airbyte"
+[[tool.poetry.packages]]
+include = "source_openweather"
+
+[tool.poetry.dependencies]
+python = "^3.9,<3.12"
+airbyte-cdk = "^0.74.0"
```

Comment:
```suggestion
airbyte-cdk = "^0"
```

Additionally, you should call out that the version of CDK should be unpinned unless you intentionally want to pin a specific one for the connector. In the future, all low-code sources will drop their pyproject.toml files and always run on latest CDK.